halpybot/packages/ircclient/halpybot.py

```diff
@@ -182,14 +185,34 @@ async def part(self, channel, message=None):
 
         """
         await super().part(channel, message)
-        chlist = config["Channels"]["channellist"].split()
+        chlist = config.channels.channel_list
         if channel in chlist:
             chlist.remove(channel)
-            config["Channels"]["channellist"] = " ".join(chlist)
+            config.channels.channel_list = " ".join(chlist)
             with open("config/config.ini", "w", encoding="UTF-8") as conf:
                 config.write(conf)
 
 
+# dynamically determine which auth method to use.
+if isinstance(config.irc.sasl, SaslExternal):
+    auth_kwargs = dict(sasl_mechanism="EXTERNAL", tls_client_cert=config.irc.sasl.cert)
+elif isinstance(config.irc.sasl, SaslPlain):
+    auth_kwargs = dict(
+        sasl_username=config.irc.sasl.username,
+        sasl_password=config.irc.sasl.password.get_secret_value(),
+    )
+else:
+    raise AssertionError(
+        f"unreachable SASL auth variant reached: {type(config.irc.sasl)}"
+    )
+client = HalpyBOT(
+    nickname=config.irc.nickname,
+    sasl_identity=config.irc.sasl.identity,
+    **auth_kwargs,
+    eventloop=asyncio.get_event_loop(),
+)
+
+
```

Comment:
This section is a modified version of an out-of-date branch. 

The previous investigation revealed that the branch was off of upstream MASTER not upstream DEVELOP.